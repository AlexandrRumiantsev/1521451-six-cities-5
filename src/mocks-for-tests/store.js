import configureStore from "redux-mock-store";
import {offers, cities, reviews, user, baseFilter} from "./mocks";

const createStore = configureStore();

const mockStore = createStore({
  Offers: {
    currentCity: cities[`Brussels`],
    baseFilter,
    offers,
    offer: offers[0],
    nearby: offers,
    favorites: offers,
    data: offers,
    getFavoriteDispatch: ()=>{}
  },
  Review: {
    reviews
  },
  User: user,
  offer: offers[0]
});

export {mockStore};
