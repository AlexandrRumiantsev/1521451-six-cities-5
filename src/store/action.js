export const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_OFFERS: `GET_OFFERS`
};

export const ActionCreator = {
  changeCity: (titleCity, cb) => ({
    type: ActionType.CHANGE_CITY,
    payload: {
      title: titleCity,
      callback: cb
    }
  }),
  getOffers: () => ({
    type: ActionType.GET_OFFERS,
    payload: 1,
  })
};

