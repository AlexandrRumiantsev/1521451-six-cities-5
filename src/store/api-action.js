export const fetch = () => (dispatch, _getState, api) => (
  api.get(`https://5.react.pages.academy/six-cities/hotels`)
    .then(({data}) => dispatch({
      type: `loadOffers`,
      payload: data
    }))
);
