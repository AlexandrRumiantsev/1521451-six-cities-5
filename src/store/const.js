export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  RESULT: `/`,
  LOGIN: `/login`
};

const baseUrl = `https://5.react.pages.academy/six-cities/`;

export const APIRoute = {
  GET_OFFER: baseUrl + `hotels`,
  LOGIN: baseUrl + `login`
};

export const actionFilter = {
  FILTER_POPULAR: `Popular`,
  FILTER_TOP_RATED_FIRST: `Top rated first`,
  FILTER_PRICE_LOW_TO_HIGH: `Price: low to high`,
  FILTER_PRICE_HIGH_TO_LOW: `Price: high to low`,
};

export const actionCity = {
  CHANGE_CITY: `CHANGE_CITY`,
  LOAD_OFFERS: `LOAD_OFFERS`
};
