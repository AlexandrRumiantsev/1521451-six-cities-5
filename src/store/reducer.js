import {ActionType} from "./action";
import {extend} from "../helpers/extend";

import offers from "../mocks/offers";
import cities from "../mocks/cities";
import reviews from "../mocks/reviews";

const initialState = {
  currentCity: cities[0].title,
  offers,
  cities,
  reviews
};

const reducer = (state = initialState, action) => {

  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return extend(state, {
        currentCity: action.payload.title,
      });
  }

  return state;
};


export {reducer};
