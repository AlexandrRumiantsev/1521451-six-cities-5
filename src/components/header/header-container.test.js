
import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {
  BrowserRouter
} from 'react-router-dom';

import {mockStore} from "../../mocks-for-tests/store";
import HeaderContainer from "./header-container";


describe(`Should HeaderContainer render correctly`, () => {


  let headerScreenComponent = null;

  beforeEach(() => {

    mockStore.dispatch = jest.fn();

    headerScreenComponent = renderer.create(
        <BrowserRouter>
          <Provider store={mockStore}>
            <HeaderContainer/>
          </Provider>
        </BrowserRouter>
    );
  });

  it(`Should headerScreenComponent connected to store render correctly`, () => {
    expect(headerScreenComponent.toJSON()).toMatchSnapshot();
  });

});
