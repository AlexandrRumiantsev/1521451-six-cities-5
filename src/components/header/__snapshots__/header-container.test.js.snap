// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should HeaderContainer render correctly Should headerScreenComponent connected to store render correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-linkheader__logo-link--active"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <div
              className="header__nav-link header__nav-link--profile"
            >
              <img
                className="header__avatar-wrapper"
              />
              <span
                className="header__user-name user__name"
              >
                <a
                  href="/favorites"
                  onClick={[Function]}
                >
                   
                   
                </a>
              </span>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
